aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -3399,15 +3407,17 @@ private void snapshotState(
 
         snapshotTaker.markBegin(SNAPSHOT_TYPE_ID, logPosition, leadershipTermId, 0, clusterTimeUnit, ctx.appVersion());
 
+        final long actualNextSessionId = role == Cluster.Role.LEADER ? nextCommittedSessionId : nextSessionId;
         if (pendingServiceMessageTrackers.length > 0)
         {
             final PendingServiceMessageTracker trackerOne = pendingServiceMessageTrackers[0];
             snapshotTaker.snapshotConsensusModuleState(
-                nextSessionId, trackerOne.nextServiceSessionId(), trackerOne.logServiceSessionId(), trackerOne.size());
+                actualNextSessionId, trackerOne.nextServiceSessionId(),
```

Comment:
Simply use `nextCommittedSessionId` which is guaranteed to be correct for any node role.